### PR TITLE
Forms - pt 1

### PR DESCRIPTION
**Pull request by MTTParry**
GETs work, DELETEs work. Lists are accessible, forms are accessible.

The PUTs don't work (see formsContinued branch) and at least the Bills' POST doesn't work.

### CODE REVIEW

File: client/src/components/Bills/BillForm.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+  notes: "",
+  full_total: 0,
+};
+
+const BillForm = (props) => {
+  //An initial student if there is one in props
+  const { initialBill } = props;
+
+  // Initial State
+  const [bill, setBill] = useState(initialBill || emptyBill);
```
Comment (sfurlani) [inline client/src/components/Bills/BillForm.js:22]: 🍪

File: client/src/components/Bills/BillForm.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
+import { useState } from "react";
+import ContactDropDown from "../DropDowns/ContactDropList";
+
+const emptyBill = {
```
Comment (sfurlani) [inline client/src/components/Bills/BillForm.js:4]: 💡 Maybe add an `emptyBill = Object.freeze({ ....} )`?

File: client/src/components/Bills/BillForm.js [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(existingBill),
+    })
+      .then((response) => {
+        return response.json();
+      })
+      .then((data) => {
+        console.log("The updated bill info: ", data);
+        props.updateContact(data);
+      });
```
Comment (sfurlani) [inline client/src/components/Bills/BillForm.js:61]: ❗ this should (in real code) have a `catch()` clause to capture errors.

File: client/src/components/Bills/BillList.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
+import React from "react";
+import { useState, useEffect } from "react";
+import BillForm from "./BillForm";
+
+const emptyBill = {
```
Comment (sfurlani) [inline client/src/components/Bills/BillList.js:5]: ❗ this is duplicated from `BillForm` - you could export from the bill form:

```

export default BillForm;
export { emptyBill };

```


```
import BillForm { emptyBill } from "./BillForm";
```

File: client/src/components/Bills/BillList.js [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+  who_paid: null,
+  paid_up: null,
+  bill_notes: "",
+  full_total: null,
+};
+
+function BillList() {
+  //This needs an empty array, or the whole thing breaks
+  const [bills, setBills] = useState([]);
+  const [editBillById, setEditBillById] = useState(null);
```
Comment (sfurlani) [inline client/src/components/Bills/BillList.js:21]: 💡  I might call this `idToEdit`?

File: client/src/components/Bills/BillList.js [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+        <li>who paid for it</li>
+        <li>have they been paid back?</li>
+      </ul>
+
+      {bills.map((bill) => {
+        if (bill.bill_id === editBillById) {
+          return <BillForm initialPost={bill} savePost={updateBill} />;
+        } else {
+          return (
+            <div className="card" key={bill.bill_id}>
```
Comment (sfurlani) [inline client/src/components/Bills/BillList.js:90]: 💡 for cleanliness sake, I might make this it's own component?

File: client/src/components/Bills/BillList.js [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+                <li>
+                  Notes: <b>{bill.bill_notes}</b>
+                </li>
+              </ul>
+
+              <button
+                className="editbuttons"
+                key="edit_${bill.bill_id}"
+                value={bill.bill_id}
+                onClick={() => editBill(bill.bill_id)}
```
Comment (sfurlani) [inline client/src/components/Bills/BillList.js:121]: 🍪

File: client/src/components/Contacts/ContactForm.js [hunk 8]
```diff
@@ -1,24 +1,25 @@
...
 };
 
 const ContactForm = (props) => {
   //An initial student if there is one in props
-  const { initialContact = { emptyContact } } = props;
+  const { initialContact } = props;
 
   // Initial State
-  const [contact, setContact] = useState(initialContact);
+  const [contact, setContact] = useState(initialContact || emptyContact);
```
Comment (sfurlani) [inline client/src/components/Contacts/ContactForm.js:15]: 🍪

File: client/src/components/Debts/DebtForm.js [hunk 9]
```diff
@@ -0,0 +1,161 @@
+import { useState } from "react";
+import ContactDropDown from "../DropDowns/ContactDropList";
+import BillDropDown from "../DropDowns/BillDropList";
+
+const emptyDebt = {
```
Comment (sfurlani) [inline client/src/components/Debts/DebtForm.js:5]: 💡 same obj.freeze

File: client/src/components/Debts/DebtForm.js [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+  debt_paid_up: null,
+  notes: "",
+};
+
+const DebtForm = (props) => {
+  //An initial student if there is one in props
+  const { initialDebt } = props;
+
+  // Initial State
+  const [debt, setDebt] = useState(initialDebt || emptyDebt);
```
Comment (sfurlani) [inline client/src/components/Debts/DebtForm.js:19]: 🍪

**Review by sfurlani — APPROVED** [review]
im afk, so without running it i cant offer further details, but some small things and you should be good to go

File: client/src/components/Bills/BillList.js (hunk 4 above)
Comment (MTTParry) [reply client/src/components/Bills/BillList.js:5]: Turns out I only use these in the Forms, but I went ahead and exported those consts for contacts, bills, and debts

File: client/src/components/Bills/BillForm.js (hunk 2 above)
Comment (MTTParry) [reply client/src/components/Bills/BillForm.js:4]: I want the values in the emptyBill to change as the user inputs information into the form. Wouldn't Object.freeze() prevent that from happening?

File: client/src/components/Bills/BillList.js (hunk 5 above)
Comment (MTTParry) [reply client/src/components/Bills/BillList.js:21]: changed it to billIdToEdit

File: client/src/components/Bills/BillForm.js (hunk 3 above)
Comment (MTTParry) [reply client/src/components/Bills/BillForm.js:61]: Did this for the POSTs, PUTs, and handleSubmits.

**Pull request merged**